src/main/java/dao/H2Factory/H2UserDao.java

```diff
@@ -0,0 +1,116 @@
+package dao.H2Factory;
+
+import customerproductorder.models.User;
+import dao.DaoException;
+import dao.UserDaoInterface;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+class H2UserDao implements UserDaoInterface {
+
+    private final ConnectionProvider connectionProvider;
+
+    protected H2UserDao(ConnectionProvider connectionProvider) {
+        this.connectionProvider = connectionProvider;
+    }
+
+    public void create(User newUser) throws DaoException {
+        try (Connection connection = connectionProvider.getConnection()) {
+            PreparedStatement st = connection.prepareStatement("insert into "
```

Comment:
It is good that you use PreparedStatement. 

But can you give me answer why it is better than using just string concatenation instead of prepared SQL statement?????